framework/src/main/java/org/checkerframework/framework/type/AnnotatedTypeFactory.java

```diff
@@ -3951,4 +3953,64 @@ public ProcessingEnvironment getProcessingEnv() {
     public CFContext getContext() {
         return checker;
     }
+
+    static final Pattern plusConstant = Pattern.compile(" *\\+ *(-?[0-9]+)$");
+    static final Pattern minusConstant = Pattern.compile(" *- *(-?[0-9]+)$");
+
+    /**
+     * Given an expression, split it into a subexpression and a constant offset. For example:
+     *
+     * <pre>{@code
+     * "a" => <"a", "0">
+     * "a + 5" => <"a", "5">
+     * "a + -5" => <"a", "-5">
+     * "a - 5" => <"a", "-5">
+     * }</pre>
+     *
+     * There are methods that can only take as input an expression that represents a Receiver. The
+     * purpose of this is to pre-process expressions to make those methods more likely to succeed.
+     *
+     * @param expression and expression to remove a constant offset from
+     * @return a sub-expression and a constant offset
+     */
+    // TODO: generalize.  There is no reason this couldn't handle arbitrary addition and subtraction
+    // expressions, given the Index Checker's support for OffsetEquation.  That might even make its
+    // implementation simpler.
+    public static Pair<String, String> getExpressionAndOffset(String expression) {
+        String expr = expression;
+        // Should the default value be null instead of "0"?
+        String offset = "0";
```

Comment:
The default value should be `"0"`. There are two cases where this method is called and the default value is returned:
* `expression` is unparseable, and should be defaulted to top, or
* `expression` is parseable but does not contain a constant offset

In the first case, the return value of this method doesn't matter - it will be ignored anyway. The only reason to prefer `null` would be to throw a runtime exception closer to the problem if that value is used.

In the second case, `"0"` matches the current default behavior - which is to parse the expression and apply an offset of zero to it. Adding null-checking to that behavior seems unnecessary, and will make the code much more complex.